src/sources/datadog_agent/logs.rs

```diff
@@ -143,13 +148,50 @@ pub(crate) fn decode_log_body(
                                 path!("ddsource"),
                                 ddsource.clone(),
                             );
-                            namespace.insert_source_metadata(
-                                source_name,
-                                log,
-                                path!("ddtags"),
-                                path!("ddtags"),
-                                ddtags.clone(),
-                            );
+
+                            if !simplify_log_tags {
+                                namespace.insert_source_metadata(
+                                    source_name,
+                                    log,
+                                    path!("ddtags"),
+                                    path!("ddtags"),
+                                    ddtags.clone(),
+                                );
+                            } else {
```

Comment:
@bruceg is doing similar work on metrics for duplicate tags, would be worth seeing if we can reuse code here - and to make sure we handle both signals the same.